### ISSUE TITLE
ATX_DEBUG macro

### ISSUE THREAD

**Issue by ademyankov** [axiomatic-systems/Atomix#2]
Hi,
   I would like to make a suggestion for your library.

I have been using Atomix for almost a year, and what I found out just now that all my ATX_ASSERT macro doesn't actually do anything! Because ATX_DEBUG wasn't define!? Which to my opinion is not a very wise idea because:

1. There is a standard, well-known way of doing that;
2. How would I know that I have to use that macro unless I reverse engineer the library?
3. Now, I have to deal with extra logic in all build scripts, doing stuff like:
`if target == debug then add ATX_DEBUG to cflags and if target == release then don't add it!
`

So, long story short, I would suggest to get rid of that and use NDEBUG.
> NDEBUG is a standard macro with the semantic "Not Debug" for C89, C99, C++98, C++2003, C++2011, C++2014 standards.

BTW, the same goes for Neptune.

Thanks,
Alex.
